website/docs/r/vdc.html.markdown

```diff
@@ -0,0 +1,124 @@
+  Provides a vCloud Director Provider VDC resource. This can be used to create and delete an VDC.
+---
+
+# vcd\_vdc
+
+Provides a vCloud Director VDC resource. This can be used to create and delete a VDC.
+Requires system administrator privileges.
+
+Supported in provider *v2.2+*
+
+## Example Usage
+
+```hcl
+provider "vcd" {
+  user     = "${var.admin_user}"
+  password = "${var.admin_password}"
+  vdc      = "System"
+  url      = "https://AcmeVcd/api"
+}
+
+resource "vcd_vdc" "my-vdc" {
+  name        = "my-vdc"
+  description = "The pride of my work"
+  org         = "my-org"
+
+  allocation_model = "ReservationPool"
+  network_pool_name = "vDC1-VXLAN-NP"
+  provider_vdc_name = "vDC1"
+
+  compute_capacity {
+    cpu {
+      units     = "MHz"
+      allocated = 2048
+      limit     = 2048
+      reserved  = 2048
+      used      = 0
+      overhead  = 0
+    }
+
+    memory {
+      units     = "MB"
+      allocated = 2048
+      limit     = 2048
+      reserved  = 2048
+      used      = 0
+      overhead  = 0
+    }
+  }
+
+  storage_profile {
+    name     = "storage-name"
+    enabled  = true
+    units    = "MB"
+    limit    = 10240
+    default  = true    
+  }
+
+  is_enabled             = true
+  is_thin_provision      = true
+  uses_fast_provisioning = true
+  delete_force           = true
+  delete_recursive       = true
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional; *v2.0+*) Organization to create the VDC in, optional if defined at provider level
+* `name` - (Required) VDC name
+* `description` - (Optional) VDC friendly description
+* `allocation_model` - (Required) The allocation model used by this VDC; must be one of {AllocationVApp, AllocationPool, ReservationPool}
+* `compute_capacity` - (Required) The compute capacity allocated to this VDC.  See [Compute Capacity](#computecapacity) below for details.
+* `nic_quota` - (Optional) Maximum number of virtual NICs allowed in this VDC. Defaults to 0, which specifies an unlimited number.
+* `network_quota` - (Optional) Maximum number of network objects that can be deployed in this VDC. Defaults to 0, which means no networks can be deployed.
+* `vm_quota` - (Optional) The maximum number of VMs that can be created in this VDC. Includes deployed and undeployed VMs in vApps and vApp templates. Defaults to 0, which specifies an unlimited number.
+* `is_enabled` - (Optional) True if this VDC is enabled for use by the organization VDCs. A VDC is always enabled on creation.
+* `storage_profile` - (Required) Storage profiles supported by this VDC.  See [Storage Profile](#storageprofile) below for details.
+* `resource_guaranteed_memory` - (Optional) Percentage of allocated memory resources guaranteed to vApps deployed in this VDC. For example, if this value is 0.75, then 75% of allocated resources are guaranteed. Required when AllocationModel is AllocationVApp or AllocationPool. Value defaults to 1.0 if the element is empty.
```

Comment:
Why are we defining percentage as float? "0.75" is quite counter-intuitive compared to just "75%" when writing and reading config.

Reply:
Cause it vCD API approach. If we different design we need agree and write converter.

Reply:
GUI has percentages. It's a matter of convenience, but I may be biased here. @Didainius and @dataclouder what are your preferences?

Reply:
Decided to leave it as float.